compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/FirExpressionUtil.kt

```diff
@@ -41,8 +43,15 @@ inline val FirCall.arguments: List<FirExpression> get() = argumentList.arguments
 
 inline val FirCall.argument: FirExpression get() = argumentList.arguments.first()
 
-inline val FirCall.argumentMapping: LinkedHashMap<FirExpression, FirValueParameter>?
-    get() = (argumentList as? FirResolvedArgumentList)?.mapping
+inline val FirCall.argumentMapping: Map<FirExpression, FirValueParameter>?
```

Comment:
It depends, but if the use site is `FIR2IR` only, then I'd handle `FirResolvedArgumentList` only.

Reply:
OK, added `resolvedArgumentMapping` and use it in fir2ir